eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/MultiReaderUnifiedSet.java

```diff
@@ -433,6 +443,16 @@ public MutableShortSet collectShort(ShortFunction<? super T> shortFunction)
     @Override
     public boolean equals(Object o)
     {
+        if (o == this || o == this.delegate)
+        {
+            return true;
+        }
+
+        if (!(o instanceof Set))
+        {
+            return false;
+        }
```

Comment:
This seems like a clever optimization. How did it come up?

Reply:
@motlin I think I tried to be consistent with the other set equals and it seemed that since it was anyway a requirement that all and every set implementations (even immutable in particular) were required to implement Set for equality test to work in both direction, I thought I would add that.
Maybe this is incorrect though and we would want those to be comparable also with non-set?

Reply:
@motlin it was already there for example for `UnifiedSet`.